### PR TITLE
[Server] Added environments in Meshery

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

- [x] POST `api/integrations/environments` to create environment
- [x] GET `api/integrations/environments` to get ALL environment
- [x] GET `api/integrations/environments/{environmentId}` to get single environment
- [x] POST `api/integrations/environments/{environmentId}` to update single environment
- [X] DELETE `api/integrations/environments/{environmentId}` to delete single environment
- [x] POST `api/integrations/environments/{environmentId}/connections/{connectionId}` to add connection in environment
- [x] GET `api/integrations/environments/{environmentId}/connections` to get environment connections
- [x] DELETE `api/integrations/environments/{environmentId}/connections/{connectionId}` to delete connection from environment
- [x] Update docs

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ghat0tkach** [comment]
hey @Yashsharma1911 
 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by l5io** [comment]
🚀 Preview for commit a187a41069e9f8d6b7ee95d900261d5fc4d4b646 at: https://650a0c7d4e15e951457e981b--meshery-docs-preview.netlify.app

File: server/handlers/environments_handlers.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+// ```?pagesize={pagesize}``` Default pagesize is 20
+//
+// ```?search={environments_name}``` If search is non empty then a greedy search is performed
+//
+// ```?filter={condition}```
+// responses:
+// 	200: environments
+
+func (h *Handler) GetEnvironments(w http.ResponseWriter, req *http.Request, _ *models.Preference, _ *models.User, provider models.Provider) {
+	logrus.Infof("Environments handler called")
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:32]:
```suggestion

```

Comment (theBeginner86) [reply server/handlers/environments_handlers.go:32]: I think, you added this just for debugging....

File: server/handlers/environments_handlers.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		return
+	}
+
+	description := fmt.Sprintf("Environment %s created.", environment.Name)
+
+	h.log.Info(description)
+	w.WriteHeader(http.StatusCreated)
+}
+
+// swagger:route GET /api/integrations/environments/{id} EnvironmentAPI idDeleteEnvironmentHandler
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:109]:
```suggestion
// swagger:route DELETE /api/integrations/environments/{id} EnvironmentAPI idDeleteEnvironmentHandler
```

File: server/handlers/environments_handlers.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+// Handle GET for Environment info by ID
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:110]:
```suggestion
// Handle DELETE for Environment based on ID
```

File: server/handlers/environments_handlers.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		http.Error(w, ErrGetResult(err).Error(), http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	fmt.Fprint(w, string(resp))
+}
+
+// swagger:route PUT /api/integrations/environments/{id} PostEnvironment idUpdateEnvironmentHandler
+// Handle POST request for updating a environment
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:130]:
```suggestion
// Handle PUT request for updating a environment
```

File: server/handlers/environments_handlers.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(respJSON)
+	if err != nil {
+		h.log.Error(ErrGetResult(err))
+		http.Error(w, "Failed to write response", http.StatusInternalServerError)
+		return
+	}
+}
+
+// swagger:route GET /api/integrations/environments/{environmentID}/connections/{connectionID} EnvironmentAPI idAddConnectionToEnvironmentHandler
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:180]:
```suggestion
// swagger:route POST /api/integrations/environments/{environmentID}/connections/{connectionID} EnvironmentAPI idAddConnectionToEnvironmentHandler
```

File: server/handlers/environments_handlers.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+		h.log.Error(ErrGetResult(err))
+		http.Error(w, ErrGetResult(err).Error(), http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	fmt.Fprint(w, string(resp))
+}
+
+// swagger:route GET /api/integrations/environments/{environmentID}/connections/{connectionID} EnvironmentAPI idRemoveConnectionFromEnvironmentHandler
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:201]:
```suggestion
// swagger:route DELETE /api/integrations/environments/{environmentID}/connections/{connectionID} EnvironmentAPI idRemoveConnectionFromEnvironmentHandler
```

**Review by theBeginner86 — COMMENTED** [review]
If you can also ensure that an event is generated (to v2 API) when these operation are performed on environment.
You can take that in new PR.

**Review by theBeginner86 — APPROVED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 932fb1fe0b2eb1779f85a25668f4c8d8c36d8c19 at: https://650b41247b76cc36a774ec85--meshery-docs-preview.netlify.app

File: server/handlers/environments_handlers.go [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"io"
+	"github.com/gorilla/mux"
+
+	"github.com/layer5io/meshery/server/models"
+	"github.com/sirupsen/logrus"
```
Comment (theBeginner86) [inline server/handlers/environments_handlers.go:11]:
```suggestion

```

**Pull request merged**